### ISSUE TITLE
Error installing on ARM Docker image with io.js

### ISSUE THREAD

**Issue by techniq** [voodootikigod/node-serialport#477]
I'm trying to `npm install serialport` using Docker with an [ARM-based image](https://registry.hub.docker.com/u/strictlybusiness/beaglebone-black-iojs/) and qemu, but the node-gyp compilation is erroring

```
root@73f15e99a1de:/# npm install serialport
-
> serialport@1.6.1 install /node_modules/serialport
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
child_process: customFds option is deprecated, use stdio instead.
make: Entering directory '/node_modules/serialport/build'
  CXX(target) Release/obj.target/serialport/src/serialport.o
  CXX(target) Release/obj.target/serialport/src/serialport_unix.o
  CXX(target) Release/obj.target/serialport/src/serialport_poller.o
  SOLINK_MODULE(target) Release/obj.target/serialport.node
collect2: fatal error: vfork: Cannot allocate memory
compilation terminated.
serialport.target.mk:112: recipe for target 'Release/obj.target/serialport.node' failed
make: *** [Release/obj.target/serialport.node] Error 1
make: Leaving directory '/node_modules/serialport/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1044:12)
gyp ERR! System Linux 3.13.0-40-generic
gyp ERR! command "/usr/bin/iojs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=serialport" "--configuration=Release" "--module_name=serialport" "--version=1.6.1" "--major=1" "--minor=6" "--patch=1" "--runtime=node" "--node_abi=node-v43" "--platform=linux" "--target_platform=linux" "--arch=arm" "--target_arch=arm" "--module_main=./serialport" "--host=https://node-serialport.s3.amazonaws.com/" "--module_path=/node_modules/serialport/build/serialport/v1.6.1/Release/node-v43-linux-arm" "--remote_path=./serialport/v1.6.1/Release/" "--package_name=node-v43-linux-arm.tar.gz" "--staged_tarball=build/stage/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz" "--hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/" "--hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz"
gyp ERR! cwd /node_modules/serialport
gyp ERR! node -v v1.4.3
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/iojs rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.6.1 --major=1 --minor=6 --patch=1 --runtime=node --node_abi=node-v43 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/node_modules/serialport/build/serialport/v1.6.1/Release/node-v43-linux-arm --remote_path=./serialport/v1.6.1/Release/ --package_name=node-v43-linux-arm.tar.gz --staged_tarball=build/stage/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/node_modules/serialport/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:984:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1057:5)
node-pre-gyp ERR! System Linux 3.13.0-40-generic
node-pre-gyp ERR! command "/usr/bin/node" "/node_modules/serialport/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /node_modules/serialport
node-pre-gyp ERR! node -v v1.4.3
node-pre-gyp ERR! node-pre-gyp -v v0.5.19
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/iojs rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.6.1 --major=1 --minor=6 --patch=1 --runtime=node --node_abi=node-v43 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/node_modules/serialport/build/serialport/v1.6.1/Release/node-v43-linux-arm --remote_path=./serialport/v1.6.1/Release/ --package_name=node-v43-linux-arm.tar.gz --staged_tarball=build/stage/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.6.1/Release/node-v43-linux-arm.tar.gz' (1)
npm ERR! Linux 3.13.0-40-generic
npm ERR! argv "/usr/bin/iojs" "/usr/bin/npm" "install" "serialport"
npm ERR! node v1.4.3
npm ERR! npm  v2.6.1
npm ERR! code ELIFECYCLE

npm ERR! serialport@1.6.1 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the serialport@1.6.1 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the serialport package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls serialport
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /npm-debug.log
root@73f15e99a1de:/# 
```

I've been able to install other packages, some like onoff which also use node-gyp, so i'm not sure what the issue might be:
```
root@73f15e99a1de:/# npm ls --depth=0
/
├── bluebird@2.9.13
├── lodash@3.3.1
├── node-uuid@1.4.3
├── onoff@1.0.2
├── request@2.53.0
├── request-debug@0.1.1
├── request-promise@0.4.0
├── stompjs@2.3.3
├── winston@0.9.0
└── ws@0.7.1
```

**Comment by techniq** [comment]
I upgraded to io.js 1.5.0 with npm 2.7.0, which [adds support](https://github.com/npm/npm/blob/master/CHANGELOG.md#a-small-feature-with-big-implications) for `--node-gyp=<PATH_TO_NODE_GYP>`.  I then installed [pangyp](https://github.com/rvagg/pangyp) and tried to install `serialport` using `npm install --node-gyp=node_modules/pangyp/bin/node-gyp.js serialport`, but it failed the same way.  I then saw this [pull request](https://github.com/mapbox/node-pre-gyp/pull/151) for `node-pre-gyp`, which adds support for pangyp (environment variable via the cli argument).

Once that PR is merged, I'll try again.  I'm not sure how bundledDependenies work as serialport has  `node-pre-gyp` as one, but serialport might need to sync up with the changes to node-pre-gyp once the PR is merged.

**Comment by voodootikigod** [comment]
Just to clarify, the reason it failed was due to "collect2: fatal error: vfork: Cannot allocate memory
compilation terminated." The reason others might have worked/succeed would be that compilation has a lot to do with the volume of code being built, I haven't looked at on/off. If node-pre-gyp has an update, nudge here -- would be glad to update to the latest. 

Thanks

**Comment by techniq** [comment]
Hmm, I saw that but was also having issues building directly on my BBB (just with io.js, not Node.js) so I thought it was directly related to the VM.

I increased the memory from the default (1GB?) to 4GB and was able to `npm install serialport` successfully.  Before I close this, I'm going to do a full project build with all my packages just incase before I close this issue.

**Comment by techniq** [comment]
I just tried a full project building on my VM with 4GB and had the following error for serialport

```
> serialport@1.2.5 install /app/node_modules/bonescript/node_modules/serialport
> node-gyp rebuild

child_process: customFds option is deprecated, use stdio instead.
make: Entering directory '/app/node_modules/bonescript/node_modules/serialport/build'
  CXX(target) Release/obj.target/serialport/src/serialport.o
In file included from ../src/serialport.cpp:3:0:
../src/serialport.h:30:38: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> List(const v8::Arguments& args);
                                      ^
../src/serialport.h:31:15: error: variable or field 'EIO_List' declared void
 void EIO_List(uv_work_t* req);
               ^
../src/serialport.h:31:15: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:31:26: error: 'req' was not declared in this scope
 void EIO_List(uv_work_t* req);
                          ^
../src/serialport.h:32:20: error: variable or field 'EIO_AfterList' declared void
 void EIO_AfterList(uv_work_t* req);
                    ^
../src/serialport.h:32:20: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:32:31: error: 'req' was not declared in this scope
 void EIO_AfterList(uv_work_t* req);
                               ^
../src/serialport.h:34:38: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> Open(const v8::Arguments& args);
                                      ^
../src/serialport.h:35:15: error: variable or field 'EIO_Open' declared void
 void EIO_Open(uv_work_t* req);
               ^
../src/serialport.h:35:15: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:35:26: error: 'req' was not declared in this scope
 void EIO_Open(uv_work_t* req);
                          ^
../src/serialport.h:36:20: error: variable or field 'EIO_AfterOpen' declared void
 void EIO_AfterOpen(uv_work_t* req);
                    ^
../src/serialport.h:36:20: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:36:31: error: 'req' was not declared in this scope
 void EIO_AfterOpen(uv_work_t* req);
                               ^
In file included from ../src/serialport.cpp:3:0:
../src/serialport.h:39:39: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> Write(const v8::Arguments& args);
                                       ^
../src/serialport.h:40:16: error: variable or field 'EIO_Write' declared void
 void EIO_Write(uv_work_t* req);
                ^
../src/serialport.h:40:16: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:40:27: error: 'req' was not declared in this scope
 void EIO_Write(uv_work_t* req);
                           ^
../src/serialport.h:41:21: error: variable or field 'EIO_AfterWrite' declared void
 void EIO_AfterWrite(uv_work_t* req);
                     ^
../src/serialport.h:41:21: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:41:32: error: 'req' was not declared in this scope
 void EIO_AfterWrite(uv_work_t* req);
                                ^
../src/serialport.h:43:39: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> Close(const v8::Arguments& args);
                                       ^
../src/serialport.h:44:16: error: variable or field 'EIO_Close' declared void
 void EIO_Close(uv_work_t* req);
                ^
../src/serialport.h:44:16: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:44:27: error: 'req' was not declared in this scope
 void EIO_Close(uv_work_t* req);
                           ^
../src/serialport.h:45:21: error: variable or field 'EIO_AfterClose' declared void
 void EIO_AfterClose(uv_work_t* req);
                     ^
../src/serialport.h:45:21: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:45:32: error: 'req' was not declared in this scope
 void EIO_AfterClose(uv_work_t* req);
                                ^
../src/serialport.h:47:39: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> Flush(const v8::Arguments& args);
                                       ^
../src/serialport.h:48:16: error: variable or field 'EIO_Flush' declared void
 void EIO_Flush(uv_work_t* req);
                ^
../src/serialport.h:48:16: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:48:27: error: 'req' was not declared in this scope
 void EIO_Flush(uv_work_t* req);
                           ^
../src/serialport.h:49:21: error: variable or field 'EIO_AfterFlush' declared void
 void EIO_AfterFlush(uv_work_t* req);
                     ^
../src/serialport.h:49:21: error: 'uv_work_t' was not declared in this scope
../src/serialport.h:49:32: error: 'req' was not declared in this scope
 void EIO_AfterFlush(uv_work_t* req);
                                ^
../src/serialport.h:89:3: error: 'uv_work_t' does not name a type
   uv_work_t req;
   ^
../src/serialport.h:90:3: error: 'ngx_queue_t' does not name a type
   ngx_queue_t queue;
   ^
In file included from ../src/serialport.cpp:8:0:
../src/serialport_poller.h:10:50: error: expected class-name before '{' token
 class SerialportPoller : public node::ObjectWrap {
                                                  ^
../src/serialport_poller.h:23:46: error: 'Arguments' in namespace 'v8' does not name a type
   static v8::Handle<v8::Value> New(const v8::Arguments& args);
                                              ^
../src/serialport_poller.h:24:48: error: 'Arguments' in namespace 'v8' does not name a type
   static v8::Handle<v8::Value> Close(const v8::Arguments& args);
                                                ^
../src/serialport_poller.h:25:48: error: 'Arguments' in namespace 'v8' does not name a type
   static v8::Handle<v8::Value> Start(const v8::Arguments& args);
                                                ^
../src/serialport_poller.h:27:3: error: 'uv_poll_t' does not name a type
   uv_poll_t poll_handle_;
   ^
../src/serialport.cpp:11:1: error: 'uv_mutex_t' does not name a type
 uv_mutex_t write_queue_mutex;
 ^
../src/serialport.cpp:12:1: error: 'ngx_queue_t' does not name a type
 ngx_queue_t write_queue;
 ^
../src/serialport.cpp:14:38: error: 'Arguments' in namespace 'v8' does not name a type
 v8::Handle<v8::Value> Open(const v8::Arguments& args) {
                                      ^
In file included from /root/.node-gyp/1.5.0/src/node.h:42:0,
                 from ../src/serialport.h:5,
                 from ../src/serialport.cpp:3:
/root/.node-gyp/1.5.0/deps/v8/include/v8.h: In function 'v8::Handle<v8::Value> Open(const int&)':
/root/.node-gyp/1.5.0/deps/v8/include/v8.h:878:13: error: 'v8::HandleScope::HandleScope()' is protected
   V8_INLINE HandleScope() {}
             ^
../src/serialport.cpp:15:19: error: within this context
   v8::HandleScope scope;
                   ^
../src/serialport.cpp:17:18: error: 'write_queue_mutex' was not declared in this scope
   uv_mutex_init(&write_queue_mutex);
                  ^
../src/serialport.cpp:17:35: error: 'uv_mutex_init' was not declared in this scope
   uv_mutex_init(&write_queue_mutex);
                                   ^
../src/serialport.cpp:18:19: error: 'write_queue' was not declared in this scope
   ngx_queue_init(&write_queue);
                   ^
../src/serialport.cpp:18:30: error: 'ngx_queue_init' was not declared in this scope
   ngx_queue_init(&write_queue);
                              ^
../src/serialport.cpp:21:13: error: invalid types 'const int[int]' for array subscript
   if(!args[0]->IsString()) {
             ^
../src/serialport.cpp:22:18: error: 'class v8::HandleScope' has no member named 'Close'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("First argument must be a string"))));
                  ^
../src/serialport.cpp:22:24: error: 'ThrowException' is not a member of 'v8'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("First argument must be a string"))));
                        ^
../src/serialport.cpp:22:68: error: 'New' is not a member of 'v8::String'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("First argument must be a string"))));
                                                                    ^
../src/serialport.cpp:24:36: error: invalid types 'const int[int]' for array subscript
   v8::String::Utf8Value path(args[0]->ToString());
                                    ^
../src/serialport.cpp:27:13: error: invalid types 'const int[int]' for array subscript
   if(!args[1]->IsObject()) {
             ^
../src/serialport.cpp:28:18: error: 'class v8::HandleScope' has no member named 'Close'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Second argument must be an object"))));
                  ^
../src/serialport.cpp:28:24: error: 'ThrowException' is not a member of 'v8'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Second argument must be an object"))));
                        ^
../src/serialport.cpp:28:68: error: 'New' is not a member of 'v8::String'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Second argument must be an object"))));
                                                                    ^
../src/serialport.cpp:30:41: error: invalid types 'const int[int]' for array subscript
   v8::Local<v8::Object> options = args[1]->ToObject();
                                         ^
../src/serialport.cpp:33:13: error: invalid types 'const int[int]' for array subscript
   if(!args[2]->IsFunction()) {
             ^
../src/serialport.cpp:34:18: error: 'class v8::HandleScope' has no member named 'Close'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Third argument must be a function"))));
                  ^
../src/serialport.cpp:34:24: error: 'ThrowException' is not a member of 'v8'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Third argument must be a function"))));
                        ^
../src/serialport.cpp:34:68: error: 'New' is not a member of 'v8::String'
     return scope.Close(v8::ThrowException(v8::Exception::TypeError(v8::String::New("Third argument must be a function"))));
                                                                    ^
../src/serialport.cpp:36:41: error: invalid types 'const int[int]' for array subscript
   v8::Local<v8::Value> callback = args[2];
                                         ^
../src/serialport.cpp:41:34: error: 'New' is not a member of 'v8::String'
   baton->baudRate = options->Get(v8::String::New("baudRate"))->ToInt32()->Int32Value();
                                  ^
../src/serialport.cpp:42:34: error: 'New' is not a member of 'v8::String'
   baton->dataBits = options->Get(v8::String::New("dataBits"))->ToInt32()->Int32Value();
                                  ^
../src/serialport.cpp:43:36: error: 'New' is not a member of 'v8::String'
   baton->bufferSize = options->Get(v8::String::New("bufferSize"))->ToInt32()->Int32Value();
                                    ^
../src/serialport.cpp:44:45: error: 'New' is not a member of 'v8::String'
   baton->parity = ToParityEnum(options->Get(v8::String::New("parity"))->ToString());
                                             ^
../src/serialport.cpp:45:48: error: 'New' is not a member of 'v8::String'
   baton->stopBits = ToStopBitEnum(options->Get(v8::String::New("stopBits"))->ToNumber()->NumberValue());
                                                ^
../src/serialport.cpp:46:32: error: 'New' is not a member of 'v8::String'
   baton->rtscts = options->Get(v8::String::New("rtscts"))->ToBoolean()->BooleanValue();
                                ^
../src/serialport.cpp:47:29: error: 'New' is not a member of 'v8::String'
   baton->xon = options->Get(v8::String::New("xon"))->ToBoolean()->BooleanValue();
                             ^
../src/serialport.cpp:48:30: error: 'New' is not a member of 'v8::String'
   baton->xoff = options->Get(v8::String::New("xoff"))->ToBoolean()->BooleanValue();
                              ^
../src/serialport.cpp:49:30: error: 'New' is not a member of 'v8::String'
   baton->xany = options->Get(v8::String::New("xany"))->ToBoolean()->BooleanValue();
                              ^
../src/serialport.cpp:52:60: error: no matching function for call to 'v8::Persistent<v8::Value>::New(v8::Local<v8::Value>&)'
   baton->callback = v8::Persistent<v8::Value>::New(callback);
                                                            ^
../src/serialport.cpp:52:60: note: candidate is:
In file included from /root/.node-gyp/1.5.0/src/node.h:42:0,
                 from ../src/serialport.h:5,
                 from ../src/serialport.cpp:3:
/root/.node-gyp/1.5.0/deps/v8/include/v8.h:6462:4: note: static T* v8::PersistentBase<T>::New(v8::Isolate*, T*) [with T = v8::Value]
 T* PersistentBase<T>::New(Isolate* isolate, T* that) {
    ^
/root/.node-gyp/1.5.0/deps/v8/include/v8.h:6462:4: note:   candidate expects 2 arguments, 1 provided
../src/serialport.cpp:53:69: error: 'New' is not a member of 'v8::String'
   baton->dataCallback = v8::Persistent<v8::Value>::New(options->Get(v8::String::New("dataCallback")));
                                                                     ^
../src/serialport.cpp:54:77: error: 'New' is not a member of 'v8::String'
   baton->disconnectedCallback = v8::Persistent<v8::Value>::New(options->Get(v8::String::New("disconnectedCallback")));
                                                                             ^
../src/serialport.cpp:55:70: error: 'New' is not a member of 'v8::String'
   baton->errorCallback = v8::Persistent<v8::Value>::New(options->Get(v8::String::New("errorCallback")));
                                                                      ^
../src/serialport.cpp:57:3: error: 'uv_work_t' was not declared in this scope
   uv_work_t* req = new uv_work_t();
   ^
../src/serialport.cpp:57:14: error: 'req' was not declared in this scope
   uv_work_t* req = new uv_work_t();
              ^
../src/serialport.cpp:57:24: error: expected type-specifier before 'uv_work_t'
   uv_work_t* req = new uv_work_t();
                        ^
../src/serialport.cpp:59:33: error: 'uv_default_loop' was not declared in this scope
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                 ^
../src/serialport.cpp:59:41: error: 'EIO_Open' was not declared in this scope
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                         ^
../src/serialport.cpp:59:52: error: 'uv_after_work_cb' was not declared in this scope
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                                    ^
../src/serialport.cpp:59:82: error: 'uv_queue_work' was not declared in this scope
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                                                                  ^
../src/serialport.cpp:61:16: error: 'class v8::HandleScope' has no member named 'Close'
   return scope.Close(v8::Undefined());
                ^
../src/serialport.cpp:61:36: error: too few arguments to function 'v8::Handle<v8::Primitive> v8::Undefined(v8::Isolate*)'
   return scope.Close(v8::Undefined());
                                    ^
In file included from /root/.node-gyp/1.5.0/src/node.h:42:0,
                 from ../src/serialport.h:5,
                 from ../src/serialport.cpp:3:
/root/.node-gyp/1.5.0/deps/v8/include/v8.h:323:28: note: declared here
   friend Handle<Primitive> Undefined(Isolate* isolate);
                            ^
../src/serialport.cpp: At global scope:
../src/serialport.cpp:64:20: error: variable or field 'EIO_AfterOpen' declared void
 void EIO_AfterOpen(uv_work_t* req) {
                    ^
../src/serialport.cpp:64:20: error: 'uv_work_t' was not declared in this scope
../src/serialport.cpp:64:31: error: 'req' was not declared in this scope
 void EIO_AfterOpen(uv_work_t* req) {
                               ^
../src/serialport.cpp: In function 'v8::Handle<v8::Value> Open(const int&)':
../src/serialport.cpp:62:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
serialport.target.mk:84: recipe for target 'Release/obj.target/serialport/src/serialport.o' failed
make: *** [Release/obj.target/serialport/src/serialport.o] Error 1
make: Leaving directory '/app/node_modules/bonescript/node_modules/serialport/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1044:12)
gyp ERR! System Linux 3.13.0-40-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /app/node_modules/bonescript/node_modules/serialport
gyp ERR! node -v v1.5.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 

> serialport@1.6.1 install /app/node_modules/serialport
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
child_process: customFds option is deprecated, use stdio instead.
make: Entering directory '/app/node_modules/serialport/build'
  CXX(target) Release/obj.target/serialport/src/serialport.o
  CXX(target) Release/obj.target/serialport/src/serialport_unix.o
  CXX(target) Release/obj.target/serialport/src/serialport_poller.o
  SOLINK_MODULE(target) Release/obj.target/serialport.node
  SOLINK_MODULE(target) Release/obj.target/serialport.node: Finished
  COPY Release/serialport.node
  COPY /app/node_modules/serialport/build/serialport/v1.6.1/Release/node-v43-linux-arm/serialport.node
  TOUCH Release/obj.target/action_after_build.stamp
make: Leaving directory '/app/node_modules/serialport/build'

> serialport@1.4.6 install /app/node_modules/pn532/node_modules/serialport
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
child_process: customFds option is deprecated, use stdio instead.
make: Entering directory '/app/node_modules/pn532/node_modules/serialport/build'
  CXX(target) Release/obj.target/serialport/src/serialport.o
In file included from ../src/serialport.h:5:0,
                 from ../src/serialport.cpp:3:
../node_modules/nan/nan.h:481:19: error: 'NanNew' declared as an 'inline' variable
       v8::String::ExternalAsciiStringResource *resource) {
                   ^
../node_modules/nan/nan.h:481:19: warning: 'always_inline' attribute ignored [-Wattributes]
../node_modules/nan/nan.h:481:19: error: 'v8::Local<v8::String> NanNew' redeclared as different kind of symbol
../node_modules/nan/nan.h:475:36: note: previous declaration 'v8::Local<v8::String> NanNew(v8::String::ExternalStringResource*)'
   NAN_INLINE v8::Local<v8::String> NanNew(
                                    ^
../node_modules/nan/nan.h:481:7: error: 'ExternalAsciiStringResource' is not a member of 'v8::String'
       v8::String::ExternalAsciiStringResource *resource) {
       ^
../node_modules/nan/nan.h:481:48: error: 'resource' was not declared in this scope
       v8::String::ExternalAsciiStringResource *resource) {
                                                ^
../node_modules/nan/nan.h: In function 'bool _NanGetExternalParts(v8::Handle<v8::Value>, const char**, size_t*)':
../node_modules/nan/nan.h:2103:12: error: 'class v8::String' has no member named 'IsExternalAscii'
   if (str->IsExternalAscii()) {
            ^
../node_modules/nan/nan.h:2104:23: error: 'ExternalAsciiStringResource' in 'class v8::String' does not name a type
     const v8::String::ExternalAsciiStringResource* ext;
                       ^
../node_modules/nan/nan.h:2105:5: error: 'ext' was not declared in this scope
     ext = str->GetExternalAsciiStringResource();
     ^
../node_modules/nan/nan.h:2105:16: error: 'class v8::String' has no member named 'GetExternalAsciiStringResource'
     ext = str->GetExternalAsciiStringResource();
                ^
serialport.target.mk:86: recipe for target 'Release/obj.target/serialport/src/serialport.o' failed
make: *** [Release/obj.target/serialport/src/serialport.o] Error 1
make: Leaving directory '/app/node_modules/pn532/node_modules/serialport/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1044:12)
gyp ERR! System Linux 3.13.0-40-generic
gyp ERR! command "/usr/bin/iojs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=serialport" "--configuration=Release" "--module_name=serialport" "--version=1.4.6" "--major=1" "--minor=4" "--patch=6" "--runtime=node" "--node_abi=node-v43" "--platform=linux" "--target_platform=linux" "--arch=arm" "--target_arch=arm" "--module_main=./serialport" "--host=https://node-serialport.s3.amazonaws.com/" "--module_path=/app/node_modules/pn532/node_modules/serialport/build/serialport/v1.4.6/Release/node-v43-linux-arm" "--remote_path=./serialport/v1.4.6/Release/" "--package_name=node-v43-linux-arm.tar.gz" "--staged_tarball=build/stage/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz" "--hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/" "--hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz"
gyp ERR! cwd /app/node_modules/pn532/node_modules/serialport
gyp ERR! node -v v1.5.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/iojs rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.4.6 --major=1 --minor=4 --patch=6 --runtime=node --node_abi=node-v43 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/app/node_modules/pn532/node_modules/serialport/build/serialport/v1.4.6/Release/node-v43-linux-arm --remote_path=./serialport/v1.4.6/Release/ --package_name=node-v43-linux-arm.tar.gz --staged_tarball=build/stage/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/app/node_modules/pn532/node_modules/serialport/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:984:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1057:5)
node-pre-gyp ERR! System Linux 3.13.0-40-generic
node-pre-gyp ERR! command "/usr/bin/node" "/app/node_modules/pn532/node_modules/serialport/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /app/node_modules/pn532/node_modules/serialport
node-pre-gyp ERR! node -v v1.5.0
node-pre-gyp ERR! node-pre-gyp -v v0.5.19
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/iojs rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.4.6 --major=1 --minor=4 --patch=6 --runtime=node --node_abi=node-v43 --platform=linux --target_platform=linux --arch=arm --target_arch=arm --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/app/node_modules/pn532/node_modules/serialport/build/serialport/v1.4.6/Release/node-v43-linux-arm --remote_path=./serialport/v1.4.6/Release/ --package_name=node-v43-linux-arm.tar.gz --staged_tarball=build/stage/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.6/Release/node-v43-linux-arm.tar.gz' (1)
npm WARN optional dep failed, continuing serialport@1.4.6
npm ERR! Linux 3.13.0-40-generic
npm ERR! argv "/usr/bin/iojs" "/usr/bin/npm" "install"
npm ERR! node v1.5.0
npm ERR! npm  v2.7.0
npm ERR! code ELIFECYCLE

npm ERR! epoll@0.0.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the epoll@0.0.7 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the epoll package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls epoll
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /app/npm-debug.log
2015/03/13 15:41:30 The command [/bin/sh -c npm install -g gulp 6to5 && npm install] returned a non-zero code: 1
```

**Comment by techniq** [comment]
Turns out the issue with building my project was I have a dependency on `serialport = ^1.4.8` but also `bonescript`, which has an [optionalDependency](https://github.com/jadonk/bonescript/blob/master/package.json#L37) on `serialport = 1.2.5`.  I removed the dependency to bonescript (as I've moved on to using other libraries such as [onoff](https://github.com/fivdi/onoff) for gpio) and the build issue went away.

Sorry for the noise and thanks for the nudge about the memory issue.
